tsl/src/gapfill/interpolate.c

```diff
@@ -125,14 +125,11 @@ gapfill_fetch_sample(GapFillState *state, GapFillInterpolateColumnState *column,
 	value = heap_getattr(&tuple, 1, tupdesc, &sample->isnull);
 	if (!sample->isnull)
 	{
-		sample->time = value;
+		sample->time = gapfill_datum_get_internal(value, state->gapfill_typid);
 
 		value = heap_getattr(&tuple, 2, tupdesc, &sample->isnull);
 		if (!sample->isnull)
-		{
-			sample->time = datumCopy(sample->time, state->columns[state->time_index]->typbyval, state->columns[state->time_index]->typlen);
 			sample->value = datumCopy(value, column->base.typbyval, column->base.typlen);
```

Comment:
@svenklemm To me it would make more sense if `value` was stored according to its internal representation as well. It does not look like we ever need the Datum representation at a later point and sth of a known size will be easier to pass around. Let me know what you think.

Reply:
@Ngalstyan4  value does not have an internal representation though cause it could be any of int2/int4/int8/float4/float8

Reply:
Ah, I see, I forgot it could also be float.